src/irr/asset/CMeshManipulator.cpp

```diff
@@ -1551,6 +1551,343 @@ core::smart_refctd_ptr<ICPUBuffer> IMeshManipulator::idxBufferFromTrianglesFanTo
+        result.asMat3x4 = core::concatenateBFollowedByA(rotation, scaleMat);
+        result.asMat3x4.setTranslation(mid);
+
+        return result;
+    };
+    
+    //calc max and min projections
+    {
+        const core::vectorSIMDf firstVtxPos = mb->getPosition(0u);
+
+        for(uint32_t i = 0u; i < projCnt; i++)
+            minProjLen[i] = maxProjLen[i] = core::dot(firstVtxPos, sampleDir[i]).x; // should be better to compute it manually..
+
+        std::fill(minVtxPos, minVtxPos + projCnt, firstVtxPos);
+        std::fill(maxVtxPos, maxVtxPos + projCnt, firstVtxPos);
+
+        for (size_t i = 1u; i < mb->calcVertexCount(); i++)
+        {
+            for (uint32_t j = 0u; j < projCnt; j++)
+            {
+                float vtxProj = core::dot(mb->getPosition(i), sampleDir[j]).x;
+
+                if (vtxProj > maxProjLen[j])
+                {
+                    maxProjLen[j] = vtxProj;
+                    maxVtxPos[j] = mb->getPosition(i);
+                }
+
+                if (vtxProj < minProjLen[j])
+                {
+                    minProjLen[j] = vtxProj;
+                    minVtxPos[j] = mb->getPosition(i);
+                }
+            }
+        }
+    }
+
+    //compute size of AABB (slabs 0, 1 and 2 define AABB)
+    AABBmid = core::vectorSIMDf(minProjLen[0] + maxProjLen[0], minProjLen[1] + maxProjLen[1], minProjLen[2] + maxProjLen[2]) * 0.5f;
+    AABBlen = core::vectorSIMDf(maxProjLen[0] - minProjLen[0], maxProjLen[1] - minProjLen[1], maxProjLen[2] - minProjLen[2]);
+    AABBArea = AABBlen.x * AABBlen.y + AABBlen.x * AABBlen.z + AABBlen.y * AABBlen.z; //half box area
+
+    // Initialize the best found orientation so far to be the standard base
+    bestVal = AABBArea;
+        // b is already an identity matrix
+
+    //TODO: handle case, where vtxCnt < 26
+
+    //construct base triangle
+    {
+        //find first 2 vertices
+        uint32_t bestPairIdx = 0u;
+        float maxDistance = core::distancesquared(minVtxPos[0], maxVtxPos[0]).x;
+        for (uint32_t i = 1u; i < projCnt; i++)
+        {
+            float distance = core::distancesquared(minVtxPos[i], maxVtxPos[i]).x;
+
+            if (distance > maxDistance)
+            {
+                maxDistance = distance;
+                bestPairIdx = i;
+            }
+        }
+
+        p0 = minVtxPos[bestPairIdx];
+        p1 = maxVtxPos[bestPairIdx];
+
+        if (core::distancesquared(p0, p1).x < eps)
+        {
+            // return AABB
+            return constructOBB(core::matrix3x4SIMD(), AABBlen / 2.0f, core::vectorSIMDf());
+        }
+
+        //TODO: check it!!
+        auto pointToLineDistanceSquared = [&](const core::vectorSIMDf p0, const core::vectorSIMDf dir, const core::vectorSIMDf v)
+        {
+            _NBL_DEBUG_BREAK_IF(core::length(dir).x < eps);
+
+            const core::vectorSIMDf u = v - p0;
+            return core::lengthsquared(u - (dir * core::dot(u, dir).x)).x;
```

Comment:
the closest distance to a line (orthogonal distance) is `sin(angle) |u|`

Reply:
which given that `dir` is unit length, the above would be the length of your cross product because `cross(a,b) = |a||b|sin(angleAB)`

`length(cross(u,dir)) == sqrt(dot(u,u)-dot(u,dir)*dot(u,dir))`

if you keep track of cross product you can use it right away for the normal of your "base triangle"
fun fact, once you